pyemu/utils/geostats.py

```diff
@@ -940,6 +943,58 @@ def calc_factors_grid(
                 "spatial_reference does not have proper attributes:{0}".format(str(e))
             )
 
+        use_ppu = False
+        if try_use_ppu:
+            try:
+                from pypestutils.pestutilslib import PestUtilsLib
+                use_ppu = True
+            except Exception as e:
+                pass
+
+
+        if use_ppu:
+            print("...pypestutils detected and being used for kriging solve, trust us, you want this!")
+            ecs = self.point_data.x.values
+            ncs = self.point_data.y.values
+            zns = 1#np.ones_like(ecs,dtype=int)
+            if "zone" in self.point_data.columns:
+                zns = self.point_data.zone.values
+            ect = x.ravel()
+            nct = y.ravel()
+            znt = 1
+            if zone_array is not None:
+                znt = zone_array.ravel()
+            assert len(self.geostruct.variograms) == 1
+            v = self.geostruct.variograms[0]
+            vartype = 2
+            if isinstance(v,ExpVario):
+                pass
+            elif isinstance(v,SphVario):
+                vartype = 1
+            elif isinstance(v, GauVarioVario):
+                vartype = 3
+            else:
+                raise NotImplementedError("unsupported variogram type: {0}".format(str(type(v))))
+            krigtype = 1 #hard coded to ordinary
+            factorfiletype = 1 #hard coded to ascii for now since we have to rewrite the fac file
+
+            plib = PestUtilsLib()
+            num_interp_pts = plib.calc_kriging_factors_2d(self.point_data.x.values,
+                                                          self.point_data.y.values,zns,
+                                                          x.ravel(),y.ravel(),znt,
+                                                          vartype,krigtype,
+                                                          v.a,v.anisotropy,v.bearing,
+                                                          search_radius,maxpts_interp,
+                                                          minpts_interp,ppu_factor_filename,
+                                                          factorfiletype)
+            plib.free_all_memory()
+            assert os.path.exists(ppu_factor_filename)
+            reformat_factorfile(x.shape[0], x.shape[1], self.point_data, self.geostruct, ppu_factor_filename)
```

Comment:
is the reformatting for downstream compatibility? with our current pyemu methods? pestutils? is there anything we can do about that?

Reply:
I tried to find a way around this but the new fac file format coming out of ppu doesn't have array shape info in the header so I thought it was easiest to patch that here when we know the shape.